armsr: armv7: add STMicroelectronics boards

jetm opened the pull request:
Add missing STMicroelectronics ARM SystemReady certificated board.

Kernel configuration added from https://github.com/STMicroelectronics/device-stm-kernel and adjusted for Linux v6.1.

mcbridematt commented:
Many thanks for doing this! I've been curious about these ST SoCs for a while now.

The config patch selects a few options already selected by either the armsr generic config or the included modules, can you check and remove ones that are already defined elsewhere?

If there is a regression in the future, it will make it easier to know what specifically the ST SoC needs to work.

jetm commented:
> Many thanks for doing this! I've been curious about these ST SoCs for a while now.

Thank you for taking a look.
 
> The config patch selects a few options already selected by either the armsr generic config or the included modules, can you check and remove ones that are already defined elsewhere?

Done. I have removed the common between generic and `armv7`:
```
$ comm -12 <(sort target/linux/armsr/config-6.1) <(sort target/linux/armsr/armv7/config-6.1)
CONFIG_CRYPTO_LIB_BLAKE2S_GENERIC=y                                                                                                                                                                                                                           
CONFIG_HZ_PERIODIC=y                                                                                                                                                                                                                                          
# CONFIG_I2C_SLAVE_TESTUNIT is not set                                                                                                                                                                                                                        
CONFIG_SERIAL_OF_PLATFORM=y
```

> If there is a regression in the future, it will make it easier to know what specifically the ST SoC needs to work.

Yes, it makes sense. Thanks!

I have another commit in queue to move other common configs between `armv7` and `armv8` to the generic one:
```
$ comm -12 <(sort target/linux/armsr/armv8/config-6.1) <(sort target/linux/armsr/armv7/config-6.1)
CONFIG_CRYPTO_SHA1=y
CONFIG_DMA_SHARED_BUFFER=y
CONFIG_GPIO_GENERIC_PLATFORM=y
CONFIG_GPIO_GENERIC=y
CONFIG_I2C_BOARDINFO=y
CONFIG_I2C=y
CONFIG_MFD_CORE=y
CONFIG_MMC_ARMMMCI=y
CONFIG_RTC_I2C_AND_SPI=y
CONFIG_USB_OHCI_HCD_PLATFORM=y
CONFIG_USB_OHCI_HCD=y
```

linusw commented on target/linux/armsr/armv7/config-6.1:
```diff
@@ -1,17 +1,26 @@
 CONFIG_ARCH_OPTIONAL_KERNEL_RWX_DEFAULT=y
 CONFIG_ARCH_SELECT_MEMORY_MODEL=y
 CONFIG_ARCH_VIRT=y
+CONFIG_ARCH_STI=y
```
Why are you activating ARCH_STI? This is used in set-top boxed and Starlink, none used by OpenWrt I think.

linusw commented on target/linux/armsr/armv7/config-6.1:
```diff
@@ -42,36 +63,216 @@ CONFIG_CPU_SPECTRE=y
+CONFIG_PCS_XPCS=y
 CONFIG_PERF_USE_VMALLOC=y
+CONFIG_PINCTRL=y
+CONFIG_PINCTRL_ST=y
```
This is also related to ARCH_STI which I think you don't want.

jetm replied:
Yes, the certificated board is a STM32. I have removed those. Thanks!

linusw commented on target/linux/armsr/armv7/config-6.1:
```diff
@@ -1,17 +1,26 @@
 CONFIG_ARCH_OPTIONAL_KERNEL_RWX_DEFAULT=y
 CONFIG_ARCH_SELECT_MEMORY_MODEL=y
 CONFIG_ARCH_VIRT=y
+CONFIG_ARCH_STI=y
```
Why are you activating ARCH_STI? This is used in set-top boxed and Starlink, none used by OpenWrt I think.

jetm replied:
Removed. Thanks!

linusw reviewed (changes requested):
Some ARCH_STI noise needs to be dropped, I think you only want ARCH_STM32?

linusw commented on target/linux/armsr/armv7/config-6.1:
```diff
@@ -1,17 +1,25 @@
+CONFIG_ARM_GT_INITIAL_PRESCALER_VAL=1
 CONFIG_ARM=y
 CONFIG_ARM_CPU_SUSPEND=y
+CONFIG_ARM_ERRATA_754322=y
```
Why are you selecteding this errata? It is selected by STI but not STM32.

jetm replied:
Removed. Thanks!

linusw commented on target/linux/armsr/armv7/config-6.1:
```diff
@@ -1,17 +1,25 @@
 CONFIG_ARM=y
 CONFIG_ARM_CPU_SUSPEND=y
+CONFIG_ARM_ERRATA_754322=y
+CONFIG_ARM_ERRATA_764369=y
```
Dito

jetm replied:
Done

linusw commented on target/linux/armsr/armv7/config-6.1:
```diff
@@ -1,17 +1,25 @@
 CONFIG_ARM_CPU_SUSPEND=y
+CONFIG_ARM_ERRATA_754322=y
+CONFIG_ARM_ERRATA_764369=y
+CONFIG_ARM_ERRATA_775420=y
```
Dito

jetm replied:
Done

linusw commented on target/linux/armsr/armv7/config-6.1:
```diff
@@ -21,13 +29,25 @@ CONFIG_ARM_LPAE=y
 CONFIG_AUTO_ZRELADDR=y
 CONFIG_BINFMT_FLAT_ARGVP_ENVP_ON_STACK=y
 CONFIG_CACHE_L2X0=y
+CONFIG_CLKSRC_ARM_GLOBAL_TIMER_SCHED_CLOCK=y
```
Why are you selecting this? I don't thing STM32 needs this or has this timer, but please explain I might be wrong.

jetm replied:
Removed

linusw commented on target/linux/armsr/armv7/config-6.1:
```diff
@@ -21,13 +29,25 @@ CONFIG_ARM_LPAE=y
+CONFIG_CLKSRC_ARM_GLOBAL_TIMER_SCHED_CLOCK=y
+CONFIG_CLKSRC_MMIO=y
+CONFIG_CLKSRC_STM32=y
+CONFIG_CLKSRC_ST_LPC=y
```
Drop this, only used by STI

jetm replied:
Done. Thanks

linusw commented on target/linux/armsr/armv7/config-6.1:
```diff
@@ -42,36 +62,205 @@ CONFIG_CPU_SPECTRE=y
 CONFIG_HAVE_SMP=y
+CONFIG_HW_RANDOM=y
+CONFIG_HW_RANDOM_OPTEE=y
+CONFIG_HW_RANDOM_ST=y
```
I think this one is for STI (not sure but I think so?)

jetm replied:
Yes, you're right. Thanks!

linusw commented on target/linux/armsr/armv7/config-6.1:
```diff
@@ -42,36 +62,205 @@ CONFIG_CPU_SPECTRE=y
+CONFIG_MMC_SDHCI_IO_ACCESSORS=y
+CONFIG_MMC_SDHCI_PCI=y
+CONFIG_MMC_SDHCI_PLTFM=y
+CONFIG_MMC_SDHCI_ST=y
```
I think you can drop all the SDHCI config options, AFAIK they are only used on STI.

jetm replied:
Remove them. Thanks

linusw commented on target/linux/armsr/armv7/config-6.1:
```diff
@@ -42,36 +62,205 @@ CONFIG_CPU_SPECTRE=y
+CONFIG_PINCTRL_STM32=y
+CONFIG_PINCTRL_STM32MP135=y
+CONFIG_PINCTRL_STM32MP157=y
+CONFIG_PL310_ERRATA_753970=y
```
Drop this, only used by STI

jetm replied:
Done

linusw commented on target/linux/armsr/armv7/config-6.1:
```diff
@@ -42,36 +62,205 @@ CONFIG_CPU_SPECTRE=y
+CONFIG_PINCTRL_STM32MP135=y
+CONFIG_PINCTRL_STM32MP157=y
+CONFIG_PL310_ERRATA_753970=y
+CONFIG_PL310_ERRATA_769419=y
```
Dito

jetm replied:
Remove them. Thanks

linusw commented on target/linux/armsr/armv7/config-6.1:
```diff
@@ -42,36 +62,205 @@ CONFIG_CPU_SPECTRE=y
+CONFIG_STM32_EXTI=y
+CONFIG_STM32_THERMAL=y
+CONFIG_STM32_WATCHDOG=y
+CONFIG_ST_IRQCHIP=y
```
Drop this, only for STI.

jetm replied:
Done

linusw commented on target/linux/armsr/armv7/config-6.1:
```diff
@@ -42,36 +62,205 @@ CONFIG_CPU_SPECTRE=y
+CONFIG_STM32_WATCHDOG=y
+CONFIG_ST_IRQCHIP=y
+# CONFIG_ST_LPC_WATCHDOG is not set
+CONFIG_ST_THERMAL=y
```
Drop this, only STI, but why is CONFIG_STM32_THERMAL not selected? I can't find it!

jetm replied:
`CONFIG_STM32_THERMAL` is selected, just before `CONFIG_STM32_WATCHDOG`. Removed the rest. Thanks

linusw commented on target/linux/armsr/armv7/config-6.1:
```diff
@@ -42,36 +62,205 @@ CONFIG_CPU_SPECTRE=y
+CONFIG_STM32_RPROC=y
+CONFIG_STM32_TIMER_CNT=y
+CONFIG_STMMAC_ETH=y
+CONFIG_ST_THERMAL_MEMMAP=y
```
Drop this, only STI.

jetm replied:
Done

linusw reviewed (changes requested):
I found a bunch of other STI-related configs when I looked closer, please check these and see if you really need them or they can be dropped. The erratas have sometimes severe performance impact so check those really close.

linusw reviewed (approved):
Looks good to me!

hauke commented on target/linux/armsr/armv7/config-6.1:
```diff
@@ -42,36 +55,188 @@ CONFIG_CPU_SPECTRE=y
+# CONFIG_INTEL_QEP is not set
+# CONFIG_INTERRUPT_CNT is not set
+# CONFIG_MICROCHIP_TCB_CAPTURE is not set
+CONFIG_JFFS2_FS=y
```
Why do you need `CONFIG_JFFS2_FS`?

jetm replied:
It's not required. I have disabled `CONFIG_JFFS2_FS`. Thanks!

hauke commented on target/linux/armsr/armv7/config-6.1:
```diff
@@ -42,36 +55,188 @@ CONFIG_CPU_SPECTRE=y
+CONFIG_DMA_VIRTUAL_CHANNELS=y
+CONFIG_DMA_SHARED_BUFFER=y
+CONFIG_DRM_STM=y
+CONFIG_DRM_STM_DSI=y
```
Do you really need DRM support in the kernel?
Is this needed at all in OpenWrt? Why not put it into a kernel module?

jetm replied:
There is no requirement for now. I have removed the DRM support. Thanks

hauke commented on target/linux/armsr/armv7/config-6.1:
```diff
@@ -42,36 +55,188 @@ CONFIG_CPU_SPECTRE=y
+CONFIG_SERIAL_STM32=y
+CONFIG_SERIAL_STM32_CONSOLE=y
 CONFIG_SMP_ON_UP=y
+CONFIG_SND=y
```
I would prefer if sound is in a kmod package

jetm replied:
I have removed the SND support, as it's not required for now. In the future, the already defined `kmod-sound-core` could be added as kmod package.

Commit:
armsr: armv7: add STMicroelectronics boards

Add missing STMicroelectronics ARM SystemReady certificated board.

Kernel configuration added from
https://github.com/STMicroelectronics/device-stm-kernel and adjusted for
Linux v6.1.

Signed-off-by: Javier Tia <javier.tia@linaro.org>

hauke commented:
Please refresh the configuration using this command: `make kernel_oldconfig CONFIG_TARGET=subtarget

jetm commented:
Closing this PR in favor of https://github.com/openwrt/openwrt/pull/13684 as armv7, it's out of scope for SystemReady Certifications boards.
